codes/go/chapter_divide_and_conquer/binary_search_recur.go

```diff
@@ -0,0 +1,31 @@
+// File: binary_search_recur.go
+// Created Time: 2023-07-19
+// Author: hongyun-robot (1836017030@qq.com)
+package chapter_divide_and_conquer
+
+func dfs(nums []int, target, i, j int) int {
```

Comment:
Add function header comment. Please refer to `codes/go/chapter_backtracking/permutations_i.go`

Reply:
The format is /* … */